addons/web_editor/static/src/js/rte.js

```diff
@@ -534,8 +534,8 @@ var RTE = Widget.extend({
             this.$last = null;
         }
         if ($editable.length && (!this.$last || this.$last[0] !== $editable[0]) &&
-                ($target.closest('[contenteditable]').attr('contenteditable') || "").toLowerCase() !== 'false') {
-
+                ($target.closest('[contenteditable]').attr('contenteditable') || "").toLowerCase() !== 'false' ||
+                (dom.isImg($target[0]) && (!this.$last || this.$last[0] !== $editable[0]))) {
```

Comment:
if image added first time in image gallery snippet then it is not focused or not selectable so to select the image this condition is added and very last condition is in and operation with this:{ dom.isImg($target[0]) } otherwise this condition will always true and it will go in infinite.